account-dao/src/main/java/com/bobocode/dao/AccountDaoImpl.java

```diff
@@ -14,32 +19,73 @@ public AccountDaoImpl(EntityManagerFactory emf) {
 
     @Override
     public void save(Account account) {
-        throw new UnsupportedOperationException("I don't wanna work without implementation!"); // todo
+        performWithinPersistenceContext(em -> em.persist(account));
     }
 
     @Override
     public Account findById(Long id) {
-        throw new UnsupportedOperationException("I don't wanna work without implementation!"); // todo
+        return performReturningWithinPersistenceContext(entityManager -> entityManager.find(Account.class, id));
     }
 
     @Override
     public Account findByEmail(String email) {
-        throw new UnsupportedOperationException("I don't wanna work without implementation!"); // todo
+        return performReturningWithinPersistenceContext(entityManager -> {
+            TypedQuery<Account> findByEmailQuery
+                    = entityManager.createQuery("select a from Account a where a.email = :email", Account.class);
+            findByEmailQuery.setParameter("email", email);
+            return findByEmailQuery.getSingleResult();
+        });
     }
 
     @Override
     public List<Account> findAll() {
-        throw new UnsupportedOperationException("I don't wanna work without implementation!"); // todo
+        return performReturningWithinPersistenceContext(entityManager ->
+                entityManager.createQuery("select a from Account a", Account.class).getResultList());
     }
 
     @Override
     public void update(Account account) {
-        throw new UnsupportedOperationException("I don't wanna work without implementation!"); // todo
+        performWithinPersistenceContext(em -> em.merge(account));
```

Comment:
Since provided **account is in the *detached state*** (it has an id, and is not loaded to the *Persistence Context*), ***Hibernate* will call the database to load this account by its id** first. Then it will **copy data from account you provided onto loaded account**. So, if the account you provided has some fields changed, these **changes will be applied to the loaded account**, and since loaded account is changed in the scope of a persistence context, ***dirty checking* will call database to perform an update**.